sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -119,6 +119,16 @@ private[sql] class JacksonGenerator(
       (row: SpecializedGetters, ordinal: Int) =>
         gen.writeNumber(row.getDouble(ordinal))
 
+    case CalendarIntervalType =>
```

Comment:
let's deal with json in another PR.

Reply:
I guess this comment https://github.com/apache/spark/pull/26102#discussion_r344585132 is valid for JSON datasource as well.

Reply:
Why wasn't this comment addressed?